lib/src/main/java/net/mossol/bot/storage/impl/MenuStorageServiceImpl.java

```diff
@@ -0,0 +1,257 @@
+            if (menu == null)  {
+                log.warn("Drink Menu Watch Failed");
+                return;
+            }
+            log.info("Drink Menu Updated : " + menu);
+            drinkMenu = menu;
+        });
+
+        try {
+            drinkMenuWatcher.awaitInitialValue(5, TimeUnit.SECONDS);
+        } catch (TimeoutException e) {
+            log.error("Failed fetch Drink Menu from Central Dogma; Set the Default Menu", e);
+        }
+
+        koreaMenu = locationInfoMongoDBRepository.findAll().stream()
+                                                 .collect(Collectors.toMap(LocationInfo::getTitle,
+                                                                           Function.identity()));
+
+        if (koreaMenu == null) {
+            log.info("Failed fetch Korea Menu from MongoDB; Set the Default Menu");
+            koreaMenu = koreaDefaultCandidate;
+        }
+    }
+
+    @Override
+    public Map<String, LocationInfo> getMenu(FoodType type) {
+        switch (type) {
+            case JAPAN_FOOD:
+                return japanMenu;
+            case KOREA_FOOD:
+                return koreaMenu;
+            case DRINK_FOOD:
+                return drinkMenu;
+            default:
+                return koreaMenu;
+        }
+    }
+
+    @Override
+    public List<LocationInfo> getMenu() {
+        return locationInfoMongoDBRepository.findAll();
+    }
+
+    private void removeMenuFromMongoDB(LocationInfo locationInfo) {
+        final String locationId = locationInfo.getId();
+        log.debug("delete location; id <{}> locationInfo <{}> ", locationId, locationInfo);
+        locationInfoMongoDBRepository.deleteById(locationId);
+    }
+
+    @Override
+    public boolean removeMenu(FoodType foodType, String food) {
+        Map<String, LocationInfo> menu = getMenu(foodType);
+        if (!menu.containsKey(food)) {
+            return false;
+        }
+
+        LocationInfo locationInfo = menu.remove(food);
+
+        if (foodType == FoodType.KOREA_FOOD) {
+            removeMenuFromMongoDB(locationInfo);
+        } else {
+            addMenuToCentralDogma(menu, foodType);
+        }
+
+        return true;
+    }
+
+    @Override
+    public boolean removeMenu(FoodType foodType, LocationInfo locationInfo) {
+        Map<String, LocationInfo> menu = getMenu(foodType);
+        String food = locationInfo.getTitle();
+        if (!menu.containsKey(food)) {
+            return false;
+        }
+
+        LocationInfo removedLocation = menu.remove(food);
+        if (locationInfo.getId() != removedLocation.getId()) {
+            log.warn("Target locationInfo and stored locationInfo id is not same; "
+                     + "targetLocationInfo<{}>;removedLocationInfo<{}>", locationInfo, removedLocation);
+        }
+
+        removeMenuFromMongoDB(locationInfo);
+        return true;
+    }
+
+    private String addMenuToMongoDB(LocationInfo locationInfo) {
```

Comment:
`addMenuToMongoDB`, but paremeter is `LocationInfo`. 

Q.
Any reason to decouple `menu` (as Map) and `locationInfo`?

Reply:
Since, request argument from mossol-cms is `LocationInfo`. Let me rename `addLocationInfoToMongoDB`